modules/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.java

```diff
@@ -0,0 +1,173 @@
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.res.Configuration;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.LocaleList;
+import android.text.TextUtils;
+import android.view.View;
+
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+
+import expo.core.ExportedModule;
+import expo.core.ModuleRegistry;
+import expo.core.interfaces.ActivityProvider;
+import expo.core.interfaces.ModuleRegistryConsumer;
+import expo.core.interfaces.services.EventEmitter;
+
+import static android.content.Intent.ACTION_LOCALE_CHANGED;
+import static android.os.Build.VERSION.SDK_INT;
+import static android.os.Build.VERSION_CODES.LOLLIPOP;
+import static android.os.Build.VERSION_CODES.N;
+import static java.util.Currency.getAvailableCurrencies;
+
+
+public class LocalizationModule extends ExportedModule implements ModuleRegistryConsumer {
+
+    private LocalesBroadcastReceiver mReceiver;
+    private ModuleRegistry mModuleRegistry;
+    private EventEmitter mEventEmitter;
+
+    public LocalizationModule(Context context) {
+        super(context);
+    }
+
+    private final Context getApplicationContext() {
+        return getCurrentActivity().getApplicationContext();
+    }
+
+    private Activity getCurrentActivity() {
+        ActivityProvider activityProvider = mModuleRegistry.getModule(ActivityProvider.class);
+        return activityProvider.getCurrentActivity();
+    }
+
+    @Override
+    public String getName() {
+        return "ExpoLocalization";
+    }
+
+    @Override
+    public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+        if (mReceiver != null) {
+            getApplicationContext().unregisterReceiver(mReceiver);
+            mReceiver = null;
+        }
+
+        mEventEmitter = moduleRegistry.getModule(EventEmitter.class);
+        mModuleRegistry = moduleRegistry;
+
+        mReceiver = new LocalesBroadcastReceiver();
+        getApplicationContext().registerReceiver(mReceiver, new IntentFilter(ACTION_LOCALE_CHANGED));
+    }
+
+    @Override
+    public Map<String, Object> getConstants() {
+        HashMap<String, Object> constants = new HashMap<>();
+
+        ArrayList<Locale> locales = getLocales();
+        ArrayList<String> localeNames = getLocaleNames(locales);
+        Boolean isRTL = TextUtils.getLayoutDirectionFromLocale(Locale.getDefault())== View.LAYOUT_DIRECTION_RTL;
+
+        constants.put("isRTL", isRTL);
+        constants.put("locale", localeNames.get(0));
+        constants.put("locales", localeNames);
+        constants.put("timezone", TimeZone.getDefault().getID());
+        constants.put("isoCurrencyCodes", getISOCurrencyCodes());
+        constants.put("country", locales.get(0).getCountry());
+
+        return constants;
+    }
+
+    private Bundle getBundledConstants() {
```

Comment:
In my humble opinion it would be cleaner to write a generic function transforming Bundle into a Map (using eg. [get](https://developer.android.com/reference/android/os/BaseBundle.html#containsKey(java.lang.String)) and [keySet](https://developer.android.com/reference/android/os/BaseBundle.html#keySet())) than repeating the same code twice, but that's only my opinion! 🙂